src/AspNet.Security.OAuth.WorkWeixin/AspNet.Security.OAuth.WorkWeixin/WorkWeixinAuthenticationHandler.cs

```diff
@@ -0,0 +1,152 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace AspNet.Security.OAuth.WorkWeixin
+{
+    public class WorkWeixinAuthenticationHandler : OAuthHandler<WorkWeixinAuthenticationOptions>
+    {
+        public WorkWeixinAuthenticationHandler(
+        [NotNull] IOptionsMonitor<WorkWeixinAuthenticationOptions> options,
+        [NotNull] ILoggerFactory logger,
+        [NotNull] UrlEncoder encoder,
+        [NotNull] ISystemClock clock)
+        : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            (int errCode, string? userId) = await GetUserIdentityfierAsync(tokens);
+            if (errCode != 0 || string.IsNullOrEmpty(userId))
+            {
+                throw new HttpRequestException($"An error (Code:{errCode}) occurred while retrieving the user identitfier");
```

Comment:
This isn't an HTTP request exception really, it's the service returning an error code.